iroh-metrics/src/core.rs

```diff
@@ -10,30 +12,129 @@ static CORE: OnceCell<Core> = OnceCell::new();
 /// It also carries a single prometheus registry to be used by all metrics.
 #[derive(Debug, Default)]
 pub struct Core {
+    #[cfg(feature = "metrics")]
     registry: Registry,
-    metrics_map: HashMap<&'static str, Box<dyn Metric>>,
+    metrics_map: ErasedSyncSet,
+}
+/// Open Metrics [`Counter`] to measure discrete events.
+///
+/// Single monotonically increasing value metric.
+#[derive(Debug, Clone)]
+pub struct Counter {
+    /// The actual prometheus counter.
```

Comment:
Not the biggest fan of re-wrapping every metric type, and some of the others can get quite ugly, but not end of the world to do.

Reply:
will figure it out with the next one we actually add 😅